### PR TITLE
Group of all users from team assigned to a task are not removed when their access is removed [SCI-8587]

### PR DESCRIPTION
**Pull request by G-Chubinidze**
Jira ticket: [SCI-8587](https://scinote.atlassian.net/browse/SCI-8587)

### What was done
_added revoking logic for all team users_

[SCI-8587]: https://scinote.atlassian.net/browse/SCI-8587?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/controllers/access_permissions/projects_controller.rb [hunk 1]
```diff
@@ -161,6 +162,19 @@ def set_project
...
+      users.each do |user|
+        user_assignment = @project.user_assignments.find_by(user: user, team: current_team)
+
+        if user_assignment.last_with_permission?(ProjectPermissions::USERS_MANAGE, assigned: :manually)
+          raise ActiveRecord::RecordInvalid
+        end
+
+        propagate_job(user_assignment, destroy: true)
+      end
+    end
```
Comment (artoscinote) [inline app/controllers/access_permissions/projects_controller.rb:176]: This logic should go into the Project model, so this happens any time the project visibility is updated.

**Review by artoscinote — CHANGES_REQUESTED** [review]

File: app/jobs/user_assignments/project_group_un_assignment_job.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@ def perform(project)
 
         remove_users_from_my_modules(project, users)
         remove_users_from_experiments(project, users)
+        remove_users_from_projects(project, users)
```
Comment (okriuchykhin) [inline app/jobs/user_assignments/project_group_un_assignment_job.rb:14]: should be remove_users_from_project, as we have here only one project

File: app/jobs/user_assignments/project_group_un_assignment_job.rb [hunk 3]
```diff
@@ -32,5 +33,27 @@ def remove_users_from_my_modules(project, users)
       my_modules.update_all(updated_at: Time.current)
       # rubocop:enable Rails/SkipsModelValidations
     end
+
+    def remove_users_from_projects(project, users)
+      users.each do |user|
+        user_assignment = project.user_assignments.find_by(user: user)
+
+        if user_assignment.last_with_permission?(ProjectPermissions::USERS_MANAGE, assigned: :manually)
```
Comment (okriuchykhin) [inline app/jobs/user_assignments/project_group_un_assignment_job.rb:41]: not only manually assigned, but any

File: app/jobs/user_assignments/project_group_un_assignment_job.rb [hunk 4]
```diff
@@ -32,5 +33,27 @@ def remove_users_from_my_modules(project, users)
...
+          raise ActiveRecord::RecordInvalid
+        end
+
+        propagate_job(project, user_assignment, destroy: true)
+      end
+    end
+
+    def propagate_job(project, user_assignment, destroy: false)
```
Comment (okriuchykhin) [inline app/jobs/user_assignments/project_group_un_assignment_job.rb:49]: why do we need to propagate if we already removed the user from tasks and experiments?

Comment (G-Chubinidze) [reply app/jobs/user_assignments/project_group_un_assignment_job.rb:49]: i tried removing assignments the same way it is for experiments/modules but then it just stuck in a loop, im not sure why, but this way works 100%

**Pull request closed without merging**